### ISSUE TITLE
[Flaky Tests] Address flaky tests preventing successful builds execution

### ISSUE THREAD

**Issue by snasovich** [rancher/rancher#41808]
The following flaky tests were identified while reviewing recent failing builds and are in Hostbusters team's scope:
* ./scripts/entry: line 8: exec: provisioning-tests-rke: not found
* Test_Operation_MP_EtcdSnapshotCreationRestoreInPlace

SURE-6454

**Comment by jiaqiluo** [comment]
More flakey tests are identified.



<details>
<summary>test_mcapp_rollback_validation </summary>

```
=================================== FAILURES ===================================
________________________ test_mcapp_rollback_validation ________________________

admin_mc = <suite.conftest.ManagementContext object at 0xffffadd7f438>
admin_pc = <suite.conftest.ProjectContext object at 0xffffad289e80>
custom_catalog = <function custom_catalog.<locals>._create_catalog at 0xffffadcc9730>
remove_resource = <function remove_resource.<locals>._cleanup at 0xffffacb03730>
restore_rancher_version = None

    @pytest.mark.nonparallel
    def test_mcapp_rollback_validation(admin_mc, admin_pc, custom_catalog,
                                       remove_resource, restore_rancher_version):
        """Test rollback validation of multi cluster apps. This test will set the
        rancher version explicitly and attempt to rollback an app with rancher
        version requirements
        """
        # 1.6.0 uses 2.0.0-2.2.0
        # 1.6.2 uses 2.1.0-2.3.0
        c_name = random_str()
        custom_catalog(name=c_name)
    
        client = admin_mc.client
        set_server_version(client, "2.1.0")
    
        cat_ns_name = "cattle-global-data:"+c_name
    
        mcapp_data = {
            'name': random_str(),
            'templateVersionId': cat_ns_name+"-chartmuseum-1.6.0",
            'targets': [{"projectId": admin_pc.project.id}],
            'roles': ["cluster-owner", "project-member"],
            "answers": [{
                "type": "answer",
                "clusterId": None,
                "projectId": None,
                "values": {
                    "defaultImage": "true",
                    "image.repository": "chartmuseum/chartmuseum",
                    "image.tag": "v0.7.1",
                    "env.open.STORAGE": "local",
                    "gcp.secret.enabled": "false",
                    "gcp.secret.key": "credentials.json",
                    "persistence.enabled": "true",
                    "persistence.size": "10Gi",
                    "ingress.enabled": "true",
                    "ingress.hosts[0]": "xip.io",
                    "service.type": "NodePort",
                    "env.open.SHOW_ADVANCED": "false",
                    "env.open.DEPTH": "0",
                    "env.open.ALLOW_OVERWRITE": "false",
                    "env.open.AUTH_ANONYMOUS_GET": "false",
                    "env.open.DISABLE_METRICS": "true"
                }
            }]
        }
    
        # First app requires a min rancher version of 2.0 so no error
        mcapp1 = client.create_multi_cluster_app(mcapp_data)
        remove_resource(mcapp1)
        wait_for_app(admin_pc, mcapp_data['name'])
    
        mcapp1 = client.reload(mcapp1)
    
        original_rev = mcapp1.revisions().data[0].name
    
        mcapp1.templateVersionId = cat_ns_name+"-chartmuseum-1.6.2"
    
        # Upgrade the app to get a rollback version
        mcapp1 = client.update_by_id_multi_cluster_app(mcapp1.id, mcapp1)
        wait_for_app(admin_pc, mcapp_data['name'])
        mcapp1 = client.reload(mcapp1)
    
>       assert mcapp1.status.revisionId != original_rev, 'app did not upgrade'
E       AssertionError: app did not upgrade
E       assert 'mcapprevision-4gl8j' != 'mcapprevision-4gl8j'
E        +  where 'mcapprevision-4gl8j' = {'conditions': [{'lastUpdateTime': '2023-07-06T16:40:59Z', 'status': 'True', 'type': 'Installed'}], 'revisionId': 'mcapprevision-4gl8j', 'type': '/v3/schemas/multiClusterAppStatus'}.revisionId
E        +    where {'conditions': [{'lastUpdateTime': '2023-07-06T16:40:59Z', 'status': 'True', 'type': 'Installed'}], 'revisionId': 'mcapprevision-4gl8j', 'type': '/v3/schemas/multiClusterAppStatus'} = {'actions': {'addProjects': 'https://172.17.0.3:8443/v3/multiClusterApps/cattle-global-data:random-96280-1688661658?ac...gradeStrategy': {'type': '/v3/schemas/upgradeStrategy'}, 'uuid': '6670da58-91e6-4230-9eca-a0e0be7afcce', 'wait': False}.status

test_multi_cluster_app.py:618: AssertionError
```
</details>


<details>
<summary>estCRTBRoleTemplateInheritance (occurred multiple times) </summary>

```
=== RUN   TestRTBTestSuite
=== RUN   TestRTBTestSuite/TestCRTBRoleTemplateInheritance
    rtbs_test.go:254: 
        	Error Trace:	/go/src/github.com/rancher/rancher/tests/v2/integration/rtbs/rtbs_test.go:254
        	Error:      	Received unexpected error:
        	            	namespaces "auto-testns--agixt" is forbidden: User "u-wvh9c" cannot get resource "namespaces" in API group "" in the namespace "auto-testns--agixt"
        	Test:       	TestRTBTestSuite/TestCRTBRoleTemplateInheritance
--- FAIL: TestRTBTestSuite (15.84s)
    --- FAIL: TestRTBTestSuite/TestCRTBRoleTemplateInheritance (14.83s)
FAIL
FAIL	github.com/rancher/rancher/tests/v2/integration/rtbs	15.875s
```

</details>


<details>
<summary>test_perform_workload_action_read_only</summary>

```
=================================== FAILURES ===================================
____________________ test_perform_workload_action_read_only ____________________
[gw15] linux -- Python 3.6.15 /go/src/github.com/rancher/rancher/tests/integration/.tox/rancher/bin/python3

admin_mc = <suite.conftest.ManagementContext object at 0xffffb6203fd0>
admin_pc = <suite.conftest.ProjectContext object at 0xffffb504c588>
remove_resource = <function remove_resource.<locals>._cleanup at 0xffffb53426a8>
user_mc = <suite.conftest.ManagementContext object at 0xffffb5353e10>
user_factory = <function user_factory.<locals>._create_user at 0xffffb5bb8e18>

    def test_perform_workload_action_read_only(admin_mc, admin_pc, remove_resource,
                                               user_mc, user_factory):
        """Tests workload actions with a read-only user and a member user.
           Note: this test exists only in 2.5+ due to pod scheduling on the
           local cluster. Manually ensure that any changes in 2.4 and below
           work."""
        client = admin_pc.client
        project = admin_pc.project
        user = user_mc
        user_member = user_factory()
    
        ns = admin_pc.cluster.client.create_namespace(
            name=random_str(),
            projectId=project.id)
        remove_resource(ns)
    
        # Create a read-only user binding.
        prtb1 = admin_mc.client.create_project_role_template_binding(
            name="prtb-" + random_str(),
            userId=user.user.id,
            projectId=project.id,
            roleTemplateId="read-only")
        remove_resource(prtb1)
        wait_until_available(user.client, project)
    
        # Then, create a member user binding.
        prtb2 = admin_mc.client.create_project_role_template_binding(
            name="prtb-" + random_str(),
            userId=user_member.user.id,
            projectId=project.id,
            roleTemplateId="project-member")
        remove_resource(prtb2)
        wait_until_available(user_member.client, project)
        user_pc = user_project_client(user, project)
        user_member_pc = user_project_client(user_member, project)
    
        # Admin user creates the workload.
        workload_name = random_str()
        workload = client.create_workload(
            name=workload_name,
            namespaceId=ns.id,
            scale=1,
            containers=[{
                'name': 'foo',
                'image': 'rancher/mirrored-library-nginx:1.21.1-alpine',
                'env': [{
                    'name': 'FOO_KEY',
                    'value': 'FOO_VALUE',
                }]
            }])
        remove_resource(workload)
        wait_for_workload(client, workload.id, ns.id)
    
        # Admin user updates the workload to yield a rollback option.
        workload.containers = [{
            'name': 'foo',
            'image': 'rancher/mirrored-library-nginx:1.21.1-alpine',
            'env': [{
                'name': 'BAR_KEY',
                'value': 'BAR_VALUE',
            }]
        }]
        workload = client.reload(workload)
        workload = client.update_by_id_workload(workload.id, workload)
    
        workload = client.reload(workload)
        wait_for_workload(client, workload.id, ns.id)
>       original_rev_id = workload.revisions().data[0].id
E       IndexError: list index out of range

test_workloads.py:411: IndexError
=============================== warnings summary ===============================
alert_common.py:21
  /go/src/github.com/rancher/rancher/tests/integration/suite/alert_common.py:21: UserWarning: The 'environ['werkzeug.server.shutdown']' function is deprecated and will be removed in Werkzeug 2.1.
    request.environ['werkzeug.server.shutdown']()

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== short test summary info ============================
FAILED test_workloads.py::test_perform_workload_action_read_only
```

</details>



<details>
<summary>rancher-webhook not ready </summary>

```
Checking if we can skip CI because of an only UI/dashboard change
Environment variable DRONE_BUILD_EVENT is push
Not skipping CI, not an only UI/dashboard change
Starting rancher server for provisioning-tests using v1.26.6+k3s1
Waiting for Rancher to be healthy
Starting rancher server using run
Waiting up to 5 minutes for rancher-webhook deployment
rancher-webhook was not available after 61 seconds
rancher-webhook was not available after 124 seconds
rancher-webhook was not available after 187 seconds
rancher-webhook was not available after 250 seconds
rancher-webhook was not available after 302 seconds
Stopping rancher server
./scripts/provisioning-tests: line 13:    68 Terminated              run_rancher
time="2023-07-05T23:16:46Z" level=fatal msg="exit status 1"
```

</details>




<details>
<summary> test_app_mysql </summary>

```
=================================== FAILURES ===================================
________________________________ test_app_mysql ________________________________
[gw2] linux -- Python 3.6.15 /go/src/github.com/rancher/rancher/tests/integration/.tox/rancher/bin/python3

admin_pc = <suite.conftest.ProjectContext object at 0x7f5b0e98aac8>
admin_mc = <suite.conftest.ManagementContext object at 0x7f5b0ee0fac8>

    def test_app_mysql(admin_pc, admin_mc):
        client = admin_pc.client
        name = random_str()
    
        ns = admin_pc.cluster.client.create_namespace(name=random_str(),
                                                      projectId=admin_pc.
                                                      project.id)
        wait_for_template_to_be_created(admin_mc.client, "library")
        answers = {
            "defaultImage": "true",
            "image": "mysql",
            "imageTag": "5.7.14",
            "mysqlDatabase": "admin",
            "mysqlPassword": "",
            "mysqlUser": "admin",
            "persistence.enabled": "false",
            "persistence.size": "8Gi",
            "persistence.storageClass": "",
            "service.nodePort": "",
            "service.port": "3306",
            "service.type": "ClusterIP"
        }
        client.create_app(
            name=name,
            externalId="catalog://?catalog=library&template=mysql&version=1.3.1&"
                       "namespace=cattle-global-data",
            targetNamespace=ns.name,
            projectId=admin_pc.project.id,
            answers=answers
        )
>       wait_for_workload(client, ns.name, count=1)

test_app.py:40: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

client = <rancher.Client object at 0x7f5b0eb56048>
ns = 'random-829195-1688599970', timeout = 60, count = 1

    def wait_for_workload(client, ns, timeout=60, count=0):
        start = time.time()
        interval = 0.5
        workloads = client.list_workload(namespaceId=ns)
        while len(workloads.data) < count:
            if time.time() - start > timeout:
                print(workloads)
>               raise Exception('Timeout waiting for workload service')
E               Exception: Timeout waiting for workload service

test_app.py:857: Exception
----------------------------- Captured stdout call -----------------------------
{'type': 'collection', 'links': {'self': 'https://172.17.0.3:8443/v3/project/local:p-cmlx7/workloads'}, 'createTypes': {'workload': 'https://172.17.0.3:8443/v3/project/local:p-cmlx7/workloads'}, 'actions': {}, 'pagination': {'limit': 1000, 'total': 0}, 'sort': {'order': 'asc', 'reverse': 'https://172.17.0.3:8443/v3/project/local:p-cmlx7/workloads?namespaceId=random-829195-1688599970&order=desc', 'links': {'uuid': 'https://172.17.0.3:8443/v3/project/local:p-cmlx7/workloads?namespaceId=random-829195-1688599970&sort=uuid'}}, 'filters': {'created': None, 'creatorId': None, 'name': None, 'namespaceId': [{'modifier': 'eq', 'value': 'random-829195-1688599970'}], 'removed': None, 'uuid': None}, 'resourceType': 'workload', 'data': []}
=============================== warnings summary ===============================
alert_common.py:21
  /go/src/github.com/rancher/rancher/tests/integration/suite/alert_common.py:21: UserWarning: The 'environ['werkzeug.server.shutdown']' function is deprecated and will be removed in Werkzeug 2.1.
    request.environ['werkzeug.server.shutdown']()

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== short test summary info ============================
FAILED test_app.py::test_app_mysql
```

</details>

**Comment by jiaqiluo** [comment]
> The following flaky tests were identified while reviewing recent failing builds and are in Hostbusters team's scope:
> 
> * ./scripts/entry: line 8: exec: provisioning-tests-rke: not found
> * Test_Operation_MP_EtcdSnapshotCreationRestoreInPlace
> 
> SURE-6454

I could not find any of those two failures in the Drone CI build history, and it is possible that I have not tracked back enough builds to see them. But on the other side, the good news is that those two do not appear in any failed build I have checked. Also, I could not reproduce them. 

> * ./scripts/entry: line 8: exec: provisioning-tests-rke: not found

The script file `provisioning-tests-rke` used to exist. It was renamed to `provisioning-tests-rke2` [in this commit](https://github.com/rancher/rancher/commit/f1eb97c565860b42c3fbb5e07e20a546516d965c#diff-73ffc28acf9dfbf3d172577a4d9b7a883279855a6071a22c11843bc46a594d84), and then removed by [this commit](https://github.com/rancher/rancher/commit/4185c14bd823c4cc860c3a2d749d0231adc71a46).  My best guess is that the failed CI did not get the full changes from the upstream repo. Therefore the Drone script is out-of-date and instructs to run the script `provisioning-tests-rke`, but the file has been renamed `provisioning-tests-rke2`. 

> * Test_Operation_MP_EtcdSnapshotCreationRestoreInPlace

As mentioned above, I could not find the failed build for this test case. But I found [another build](https://drone-publish.rancher.io/rancher/rancher/9930/3/2) which failed at a similar test case: [Test_Operation_MP_EtcdSnapshotOperationsOnNewNode](https://github.com/rancher/rancher/blob/de83b5c9737cb45f8d2ddb787e82651c629c50c2/tests/v2prov/tests/machineprovisioning/operation_etcdsnapshot_newnode_test.go#L25-L113). 


I do not find any logical error in our code around those two test cases. But I am making some generate improvements to the error messages and the output for troubleshooting to hopefully make it easier for us to troubleshoot CI failures.
